main.py

```diff
@@ -0,0 +1,58 @@
+from sklearn import tree, model_selection
+import graphviz
+from sklearn import ensemble
+from sklearn import preprocessing
+import xgboost as xgb
+from sklearn.model_selection import train_test_split
+import joblib
+import pandas as pd
+
+filename = 'weights/titanic_survivor_grad_boost_classifier.pkl'
+
+
+def convert_csv_to_arrays(path_to_csv: str, train: bool):
+    original_df = pd.read_csv(path_to_csv, sep=',', header=0)\
+        .drop('Ticket', axis=1)\
+        .drop('Cabin', axis=1)
+    original_df.interpolate(inplace=True)
+
+    if train:
+        out_df = original_df['Survived'].copy()
+        in_df = original_df.drop('Survived', axis=1)
+    else:
+        in_df = original_df.copy()
+
+    in_df = in_df
+    in_df['Sex'] = in_df['Sex'].map(lambda x: 1 if x == 'male' else 0)
+    lb = preprocessing.LabelEncoder()
+    min_max_scaler = preprocessing.MaxAbsScaler()
+
+    in_df['Embarked'] = lb.fit_transform(in_df['Embarked'])
+    in_df['Name'] = lb.fit_transform(in_df['Name'])
```

Comment:
Each feature (column) usually needs its own preprocessing. For continuous numerical columns you can use sklearn's `StandardScaler`, and for categorical and numerical discrete - `OrdinalEncoder`. `LabelEncoder` is used to encode labels. For each column, you should create a preprocessor from scratch, fit it and transform the column.

Here you use the same processor object for different columns, this can cause unexpected preprocessing results.